### PR TITLE
[JBPM-8922] RemoteBusinessRuleTaskHandler does not work with https url

### PR DESCRIPTION
**Pull request by elguardian**
force https detection during load balancing.

### CODE REVIEW

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/ForceDetectUseSSLTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.kie.server.client;
```
Comment (mswiderski) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/ForceDetectUseSSLTest.java]: missing license header

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/util/SSLUtilities.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package org.kie.server.client.util;
```
Comment (mswiderski) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/util/SSLUtilities.java]: missing license header

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/KieServicesConfigurationImpl.java [hunk 3]
```diff
@@ -121,10 +121,24 @@ public KieServicesConfigurationImpl(String url, CredentialsProvider credentialsP
...
     public KieServicesConfigurationImpl(String url, CredentialsProvider credentialsProvider, long timeout) {
         this.transport = Transport.REST;
-
         this.serverUrl = url;
         this.timeoutInMillisecs = timeout;
         this.credentialsProvider = credentialsProvider;
+        //        this.useSsl = autoDetectUseSSL(url);
+    }
+
+    public boolean autoDetectUseSSL(String url) {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/KieServicesConfigurationImpl.java]: not sure I follow this method... as the useSsl seems to be used by JMS layer and now we just check the http endpoints... 

https://github.com/kiegroup/droolsjbpm-integration/blob/bc0380c7785501b53943a81f5421d10a4ecc1f3d/kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/KieServicesConfigurationImpl.java#L63

could you please explain a bit what are we trying to do here?

**Comment by elguardian** [comment]
sorry about the confusion. The problem is more how the system should behave (i messed up with the PR). The test case relevant should be https://github.com/kiegroup/droolsjbpm-integration/pull/1952/files#diff-3794651273729e747889499f86dea895R18

This test fails during kie services client constructor because the init of the server client requires at least one endpoint to be available. 

In the jira reported the https is failing because the truststore is not configured (I created a test case regarding load balancing that prooves that it works with http and https) but what about if you try to deploy while there is no endpoint available (real endpoint down) ? Not really sure if this needs to fail or not). The jira of this is not really a bug but a missconfiguration so not really sure if this was on purpose or not. @mswiderski  wdyt ?

**Comment by mswiderski** [comment]
@elguardian in general the endpoint should be available, at least one at the time of creation of kie server client. So if that is not available I'd say it's really an edge case and the kie server client itself can't do much about it.
Personally I prefer to fail fast when there are no endpoints available as it could be temporarily or it can be as simple as typo and thus way more severe.

**Pull request closed without merging**